app/controllers/PropertyLinkingController.scala

```diff
@@ -63,110 +66,34 @@ class PropertyLinkingController @Inject()(
-        params = paginationParams,
-        sortfield = sortfield,
-        sortorder = sortorder,
-        address = address,
-        agent = agent).map(x => Ok(Json.toJson(x)))
-      case None =>
-        Logger.error(s"Agent details lookup failed for agentCode: $agentCode")
-        NotFound
+
+  def getMyOrganisationsPropertyLinks(searchParams: GetPropertyLinksParameters,
+                                      paginationParams: Option[PaginationParams]) = authenticated { implicit request =>
+
+    propertyLinkService.getMyOrganisationsPropertyLinks(searchParams, paginationParams) flatMap {
+      response => {
+        Ok(Json.toJson(response))
+      }
     }
   }
 
-  /** *
-    * Make two calls to the Search/Sort API
-    * the first call returns the results based on supplied filters and sortfield
-    * the second call is used only to allow us to get the count of PENDING representation requests
-    */
-  def forAgentSearchAndSort(organisationId: Long,
-                            paginationParams: PaginationParams,
-                            sortfield: Option[String],
-                            sortorder: Option[String],
-                            status: Option[String],
-                            address: Option[String],
-                            baref: Option[String],
-                            client: Option[String]) = authenticated { implicit request =>
-    val eventualAuthResultBE = propertyLinksConnector.agentSearchAndSort(
-      organisationId = organisationId,
-      params = paginationParams,
-      sortfield = sortfield,
-      sortorder = sortorder,
-      status = status,
-      address = address,
-      baref = baref,
-      client = client,
-      representationStatus = Some("APPROVED")) // TODO cater for other statuses
-
-    // required to calculate the pending count - no filtering/sorting required
-    val eventualAuthResultPendingBE = propertyLinksConnector.agentSearchAndSort(
-      organisationId = organisationId,
-      params = paginationParams,
-      representationStatus = Some("PENDING"))
-
-    // required to get the correct filtered amount - no filtering/sorting required
-    val eventualAuthResultApprovedNoFiltersBE = propertyLinksConnector.agentSearchAndSort(
-      organisationId = organisationId,
-      params = paginationParams,
-      representationStatus = Some("APPROVED"))
-
-    for {
-      authResultBE <- eventualAuthResultBE
-      authResultPendingBE <- eventualAuthResultPendingBE
-      authResultApprovedNoFiltersBE <- eventualAuthResultApprovedNoFiltersBE
-    } yield Ok(Json.toJson(AgentAuthResultFE(authResultBE, authResultPendingBE.filterTotal, authResultApprovedNoFiltersBE.filterTotal)))
+  def getClientsPropertyLinks(searchParams: GetPropertyLinksParameters,
+                         paginationParams: Option[PaginationParams]) = authenticated { implicit request =>
 
+    propertyLinkService.getClientsPropertyLinks(searchParams, paginationParams) flatMap {
```

Comment:
does this compile?

Reply:
It compiles but happy to change if you see issues